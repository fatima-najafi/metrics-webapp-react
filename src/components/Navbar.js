import { v4 as uuidv4 } from 'uuid';
import { useNavigate } from 'react-router-dom';
import back from '../images/back.png';
import mic from '../images/mic.png';
import setting from '../images/setting.png';

const Navbar = () => {
  const navigate = useNavigate();

  const handleButtonClick = () => {
    navigate('/');
  };

  return (
    <div className="navigation" style={{ display: 'flex', gap: '20px' }}>
      <div
        role="button"
        tabIndex={0}
        key={uuidv4()}
        onClick={handleButtonClick}
        onKeyDown={(event) => {
          if (event.key === 'Enter') {
            handleButtonClick();
          }
        }}
      >
        <img
          className="btn-back"
          alt="back"
          src={back}
          style={{ backgroundColor: 'black' }}
        />
      </div>
      <h2 className="heading">Companies Webpage</h2>
      <div className="position">
        <img
          src={mic}
          alt="microphone"
          className="btn"
          style={{ backgroundColor: 'red' }}
        />
        <img
          src={setting}
          alt="setting"
          className="btn"
          style={{ backgroundColor: 'green' }}
        />
      </div>
    </div>
  );
};

export default Navbar;
