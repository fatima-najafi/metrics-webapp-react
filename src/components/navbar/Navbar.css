
.navigation {
  background-color: rgba(114, 37, 93, 0.518);
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 30px;
}

.heading {
  color: white;
  text-decoration: none;
  font-size: 18px;
  font-family: 'Lato', sans-serif;
  font-weight: 700;
}

.position {
  display: flex;
  gap: 20px;
  font-size: 20px;
}

.btn {
  width: 24px;
  height: 24px;
  cursor: pointer;
}

/* for desctop version */
@media only screen and (min-width: 768px) {
  .navigation {
    gap: 25px;
    display: flex;
    align-items: center;
  }

  .div-search {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  .heading {
    color: white;
    text-decoration: none;
    font-size: 24px;
    font-family: 'Lato', sans-serif;
    font-weight: 700;
  }

  .input {
    text-indent: 20px;
    font-family: "Poppins", sans-serif;
    border-radius: 10px;
    margin-top: 0;
    height: 40px;
    width: 500px;
    border: 2px solid rgb(255, 255, 255);
  }

  .btn {
    margin-top: -10px;
    width: 32px;
    height: 32px;
  }
}
