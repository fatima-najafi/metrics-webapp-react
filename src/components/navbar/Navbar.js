import React from 'react';
import { useNavigate } from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';
import mic from '../../images/mic.png';
import setting from '../../images/setting.png';
import back from '../../images/back.png';
import './Navbar.css';

const Navbar = () => {
  const navigate = useNavigate();

  return (
    <div className="navigation" data-testid="navbar">
      <div key={uuidv4()} onClick={() => navigate('/')} aria-hidden="true">
        <img className="btn" src={back} alt="back" />
      </div>
      <h2 className="heading">Companies List</h2>
      <div className="position">
        <img className="btn" src={mic} alt="mic" />
        <img className="btn" src={setting} alt="setting" />
      </div>
    </div>
  );
};

export default Navbar;
