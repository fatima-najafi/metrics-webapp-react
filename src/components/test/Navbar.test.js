import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter, useNavigate } from 'react-router-dom';
import Navbar from '../navbar/Navbar';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}));

describe('NavBar', () => {
  test('navigates to the root path when the back button is clicked', () => {
    const mockNavigate = jest.fn();
    useNavigate.mockReturnValue(mockNavigate);

    const { getByAltText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );

    const backButton = getByAltText('back');
    fireEvent.click(backButton);

    expect(mockNavigate).toHaveBeenCalledWith('/');
  });

  test('renders the Navbar component', () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );

    const navbar = getByTestId('navbar');
    expect(navbar).toBeTruthy();
  });
});
