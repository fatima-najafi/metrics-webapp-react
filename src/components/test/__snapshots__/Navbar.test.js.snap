// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<div>
  <div
    class="navigation"
  >
    <div
      aria-hidden="true"
    >
      <img
        alt="back"
        class="btn"
        src="back.png"
      />
    </div>
    <h2
      class="heading"
    >
      Companies List
    </h2>
    <div
      class="position"
    >
      <img
        alt="mic"
        class="btn"
        src="mic.png"
      />
      <img
        alt="setting"
        class="btn"
        src="setting.png"
      />
    </div>
  </div>
</div>
`;
